frontend/pages/SoftwarePage/SoftwareAddPage/SoftwareCustomPackage/SoftwareCustomPackage.tsx

```diff
@@ -35,7 +38,21 @@ const SoftwareCustomPackage = ({
   setSidePanelOpen,
 }: ISoftwarePackageProps) => {
   const { renderFlash } = useContext(NotificationContext);
-  const [isUploading, setIsUploading] = React.useState(false);
+  const [uploadProgress, setUploadProgress] = React.useState(0);
+  const [uploadDetails, setUploadDetails] = React.useState<IFileDetails | null>(
+    null
+  );
+
+  // TODO: This will cancel the request when the component unmounts. Notably when the back
+  // button is clicked, we haven't found a way to showing the native unload prompt. Without this effect,
+  // a long running request would continue in the background. What do we want to happen?
+  const abortRef = React.useRef(new AbortController());
+  useEffect(() => {
+    abortRef.current = new AbortController();
```

Comment:
why is this assignment needed when its done on the line outside of the useEffect?

Reply:
Yeah, it's weird and I don't think it is strictly necessary, but you get a linter warning about using it in the return if you don't set it there. I cobbled this together after searching around for patterns to use for with axios and didn't find very much guidance as most of it was focused on garden-variety axios get requests that can be entirely encapsulated in one effect rather than callback-based form submissions. I definitely think it would be worth more investigation into the patterns (probably react-query's useMutation can help with this). Based on discussion in standup, I'm going to remove the whole abort piece from this PR anyway.